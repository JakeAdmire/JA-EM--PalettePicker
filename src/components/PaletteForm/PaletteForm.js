import React, { Component } from 'react';

export class PaletteForm extends Component {

  mapProjectOptions = () => {
    const options = this.props.projects.map(project => 
      <option id={project.id} key={project.id}>{project.title}</option>);

    return options;
  }

  render() {

    return (

      <form id="palettes" onSubmit={this.props.handleSubmit} autoComplete="off">
        <label htmlFor="paletteTitle">Save this Palette:</label>
        <div>
          <select>
            { 
              this.props.projects.length 
                ? this.mapProjectOptions()
                : <option></option>
            }
          </select>
          <input  onChange={this.props.handleChange} 
                  id="paletteTitle" 
                  value={this.props.paletteTitle} 
                  type="text" 
                  placeholder={this.props.paletteError || "ex. Autumn Colors.."}>
          </input>
          <input className="submit-btn" type="submit"></input>
        </div>
      </form>

    )
  }
}