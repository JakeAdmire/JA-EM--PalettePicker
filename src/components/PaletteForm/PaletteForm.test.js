import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import { PaletteForm } from './PaletteForm';

describe('PaletteForm', () => {

  let wrapper;

  let handleSubmitMock = jest.fn();
  let handleChangeMock = jest.fn();
  let paletteTitleMock = '';
  let paletteErrorMock = '';
  let projectsMock = [
    { 
      title: "Test Project", 
      id: 73, 
      palettes: [
        {
          color1: "#7DBB54",
          color2: "#569963",
          color3: "#486A60",
          color4: "#455154",
          color5: "#313235",
          id: 63,
          name: "Test Palette",
          project_id: 73
        }
      ] 
    }
  ];

  beforeEach(() => {
    wrapper = shallow (
      <PaletteForm  handleSubmit={handleSubmitMock}
                    handleChange={handleChangeMock}
                    paletteTitle={paletteTitleMock}
                    paletteError={paletteErrorMock}
                    projects={projectsMock} />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapProjectOptions()', () => {

    it('should return JSX', () => {
      let results = wrapper.instance().mapProjectOptions();

      expect(results).toMatchSnapshot();
    })
    
  })

})