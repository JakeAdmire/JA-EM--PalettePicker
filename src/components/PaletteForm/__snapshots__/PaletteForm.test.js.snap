// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm mapProjectOptions() should return JSX 1`] = `
Array [
  <option
    id={73}
  >
    Test Project
  </option>,
]
`;

exports[`PaletteForm should match snapshot 1`] = `ShallowWrapper {}`;
